gear-tech/gear
# ci(label): trigger workflows on labelling

Resolves #2888

1. list all [check runs][0] for the PR
2. find if the latest check has processed workflow `build`
3. dispatch workflow with github API
4. create checks from the dispatched workflow
5. update the status of the checks till they are completed

- [x] trigger build on `A0-pleasereview`
- [x] dispatch workflow and create specified build checks
- [x] maintain the status of the new created checks
- [x] not trigger workflow build when it has already been triggered

---

Why create checks instead of just triggering the workflows with labels? 

Triggering the workflows directly will cause the workflows be triggered on each labelling and even if we can skip triggering on `non-A0/E2` labels, the state of the check suite will be reset which means our build checks will be disappeared as well

@gear-tech/dev 


[0]: https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#list-check-runs-for-a-git-reference
